R/utils.R

```diff
@@ -60,3 +23,23 @@ get_utm_zone_epsg <- function(x) {
   }
   base + floor((coords[1] + 180) / 6) + 1
 }
+
+#' Get UTM zone from bbox object
+#'
+#' @param bb A bbox object
+#'
+#' @return The UTM zone
+#' @export
+get_utm_zone_epsg_bbox <- function(bb) {
+  if (!inherits(bb, "bbox")) {
+    stop("the bounding box must be of class bbox")
+  }
+
+  centroid_long <- (bb[1] + bb[3]) / 2
+  centroid_lat <- (bb[2] + bb[4]) / 2
+
+  base <- if (centroid_lat >= 0) 32600 else 32700
+  utm_zone <- base + floor((centroid_long + 180) / 6) + 1
+
+  return(utm_zone)
+}
```

Comment:
What about in having a single function here working for both `sf`/`sfc` objects and  `bbox`? The following is exploiting the fact that `sf::st_bbox(x)` returns a `bbox` even if `x` is already a `bbox`. Also using "xmin", "xmax", etc. instead of the indices since it looks more explicit.

```suggestion
#' Get UTM zone as EPSG code
#'
#' @param x Bounding box or geometry object
#'
#' @return The UTM zone
#' @export
get_utm_zone <- function(x) {

  bb <- sf::st_bbox(x)
  
  centroid_long <- (bb["xmin"] + bb["xmax"]) / 2
  centroid_lat <- (bb["ymin"] + bb["ymax"]) / 2

  base <- if (centroid_lat >= 0) 32600 else 32700
  utm_zone <- base + floor((centroid_long + 180) / 6) + 1

  return(utm_zone)
}
```